cypress/plugins/index.js

```diff
@@ -1,22 +1,10 @@
-/// <reference types="cypress" />
-// ***********************************************************
-// This example plugins/index.js can be used to load plugins
-//
-// You can change the location of this file or turn off loading
-// the plugins file with the 'pluginsFile' configuration option.
-//
-// You can read more here:
-// https://on.cypress.io/plugins-guide
-// ***********************************************************
-
-// This function is called when a project is opened or re-opened (e.g. due to
-// the project's config changing)
-
-/**
- * @type {Cypress.PluginConfig}
- */
-// eslint-disable-next-line no-unused-vars
 module.exports = (on, config) => {
-  // `on` is used to hook into various events Cypress emits
-  // `config` is the resolved Cypress config
-}
+  if (config.testingType === 'component') {
+    const { startDevServer } = require('@cypress/webpack-dev-server');
+    
+    // path to your webpack.cypress config
+    const webpackConfig = require('../webpack.cypress.config');
+
+    on('dev-server:start', options => startDevServer({ options, webpackConfig }));
```

Comment:
@miram13 - is there any reason we are trying to start the server from cypress? I thought they recommended against this